### DISCUSSION TITLE
Renovate won't create pull requests for NPM packages

### DISCUSSION THREAD

**Discussion by clupo** [renovatebot/renovate#15757] [Request Help] [answered]
I'm having trouble getting Renovate to create pull requests for npm packages. I've been able to configure it to create pull requests for nuget packages but not npm

Example log of an npm package I think should be getting an update but isn't getting one:
Current value of morgan is 1.8.0 and it should get an update for 1.10.0, but the `updates` array is empty and there is no pull request created for it
```
{
  "depType": "dependencies",
  "depName": "morgan",
  "currentValue": "^1.8.0",
  "datasource": "npm",
  "prettyDepType": "dependency",
  "lockedVersion": "1.10.0",
  "depIndex": 12,
  "updates": [],
  "warnings": [],
  "versioning": "npm",
  "sourceUrl": "https://github.com/expressjs/morgan",
  "currentVersion": "1.10.0",
  "fixedVersion": "1.10.0"
},
```

Example log of a nuget packge in the same repository that's behaving as expected:
Current value of CrossBreeze.CrossTest is 1.0.0.0 and it should get an update for 1.0.2.3, the `updates` array has an element in it and there is a pull request created for it
```
{
  "datasource": "nuget",
  "depType": "nuget",
  "depName": "CrossBreeze.CrossTest",
  "currentValue": "1.0.0.0",
  "depIndex": 16,
  "updates": [
    {
      "bucket": "non-major",
      "newVersion": "1.0.2.3",
      "newValue": "1.0.2.3",
      "releaseTimestamp": "2022-03-18T15:52:29.600Z",
      "newMajor": 1,
      "newMinor": 0,
      "updateType": "patch",
      "branchName": "renovate/crossbreeze.crosstest-1.x"
    }
  ],
  "warnings": [],
  "versioning": "nuget",
  "sourceUrl": "http://www.x-test.nl/",
  "homepage": "http://www.x-test.nl/",
  "currentVersion": "1.0.0.0",
  "isSingleVersion": true,
  "fixedVersion": "1.0.0.0"
}
```

My renovate.json:
```
{
  "extends": [
    "config:base"
  ],
  "timezone": "America/New_York",
  "schedule": ["every 1 hours every weekday"]
}
```

I'm using the github Renovate app

I've tried looking through documentation, and looking at other renovate.json examples in other projects on github, and other discussion threads, but don't see much to help. Which makes me think I'm missing something obvious but I don't see what it might be

**Comment by viceice** [comment, marked as answer]
you probably using ranges, which aren't updated until range doesn't satisfy by default

**Reply by rarkins** [reply]
rangeStrategy=update-lockfile

**Reply by clupo** [reply]
Thank you!
